### PR TITLE
WT-10384 Don't allow WT_ROLLBACK for read only operations not in a transaction

### PR DESCRIPTION
**Pull request by sueloverso**
Here's the minimal change to allow this CONFIG to pass. Otherwise it fails 100% of the time for me on a 5x parallel run (that is, at least one of the 5 fail, not that all 5 fail). 

We have been seeing A LOT of mismatch failures due to this issue. There are probably a dozen or more. This change is only a partial revert of WT-8869. This change leaves in the possibility that a read-only transaction can get WT_ROLLBACK. It removes that from an operation that is not part of a transaction. So it narrows the condition.

Please review.

### CODE REVIEW

**Review by keitharnoldsmith — APPROVED** [review]
LGTM!

File: test/format/failure_configs/CONFIG.WT-10384 [hunk 1]
```diff
@@ -0,0 +1,163 @@
+[2023/01/15 23:27:10.720]     ############################################
```
Comment (etienneptl) [inline test/format/failure_configs/CONFIG.WT-10384:1]: Can you remove the timestamp at the start of each line?

**Review by etienneptl — APPROVED** [review]
Thanks for all the work on this ticket, @sueloverso!
I left one comment regarding the test/format configuration.

File: test/format/failure_configs/CONFIG.WT-10384 (hunk 1 above)
Comment (sueloverso) [reply test/format/failure_configs/CONFIG.WT-10384:1]: Yes I will do that. My eyes simply didn't see it anymore.

**Review by agorrod — APPROVED** [review]
Thanks @sueloverso thanks for finding an elegant solution - I know it was hard to thread the needle.

**Pull request merged**